### ISSUE TITLE
WGSL: vs_main returning a type that does not contain a builtin(position) causes naga to panic

### ISSUE THREAD

**Issue by rukai** [gfx-rs/naga#895]
Running wgpu master with this change to an example causes this panic in naga:
```
rukai@memes examples$ git diff
diff --git a/examples/cube/shader.wgsl b/examples/cube/shader.wgsl
index 58599f284..cef37ec4a 100644
--- a/examples/cube/shader.wgsl
+++ b/examples/cube/shader.wgsl
@@ -14,7 +14,7 @@ var r_locals: Locals;
 fn vs_main(
     [[location(0)]] position: vec4<f32>,
     [[location(1)]] tex_coord: vec2<f32>,
-) -> VertexOutput {
+) -> vec4<f32> {
     var out: VertexOutput;
     out.tex_coord = tex_coord;
     out.position = r_locals.transform * position;
rukai@memes examples$ cargo run --release --example cube
   Compiling wgpu v0.8.0 (/home/rukai2/Projects/Crates/wgpu/wgpu-rs)
    Finished release [optimized] target(s) in 1.11s
     Running `/home/rukai2/Projects/Crates/wgpu/wgpu-rs/target/release/examples/cube`
Using NVIDIA GeForce GTX 960 (Vulkan)
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /home/rukai/.cargo/git/checkouts/naga-dbb2b19faed49210/057d03a/src/proc/interpolator.rs:90:52
stack backtrace:
   0: rust_begin_unwind
             at /rustc/88f19c6dab716c6281af7602e30f413e809c5974/library/std/src/panicking.rs:493:5
   1: core::panicking::panic_fmt
             at /rustc/88f19c6dab716c6281af7602e30f413e809c5974/library/core/src/panicking.rs:92:14
   2: core::panicking::panic
             at /rustc/88f19c6dab716c6281af7602e30f413e809c5974/library/core/src/panicking.rs:50:5
   3: naga::proc::interpolator::<impl naga::Module>::apply_common_default_interpolation::default_binding_or_struct
   4: naga::front::wgsl::Parser::parse
   5: naga::front::wgsl::parse_str
   6: wgpu_core::device::Device<B>::create_shader_module
   7: wgpu_core::device::<impl wgpu_core::hub::Global<G>>::device_create_shader_module
   8: <wgpu::backend::direct::Context as wgpu::Context>::device_create_shader_module
   9: wgpu::Device::create_shader_module
  10: <cube::Example as cube::framework::Example>::init
  11: cube::framework::start
  12: cube::framework::run
  13: cube::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

I would expect an error to be reported to the user instead of panicking.

**Comment by jimblandy** [comment]
I think this should be fixed by 50de6ae (#894).

**Comment by jimblandy** [comment]
Verified, fixed by #894.

**Comment by jimblandy** [comment]
With that PR, here's the message I get:

```
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0

Using Intel(R) HD Graphics P530 (SKL GT2) (Vulkan)
[2021-05-22T20:36:07Z ERROR naga::valid::function] Returning Some(Struct { top_level: false, members: [StructMember { name: Some("tex_coord"), ty: [1], binding: Some(Location { location: 0, interpolation: Some(Perspective), sampling: Some(Center) }), offset: 0 }, StructMember { name: Some("position"), ty: [2], binding: Some(BuiltIn(Position)), offset: 16 }], span: 32 }) where Some(Vector { size: Quad, kind: Float, width: 4 }) is expected
wgpu error: Validation Error

Caused by:
    In Device::create_shader_module
    Entry point vs_main at Vertex is invalid
    The `return` value Some([10]) does not match the function return value


thread 'main' panicked at 'Handling wgpu errors as fatal by default', src/backend/direct.rs:1943:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

That seems to be a more appropriate error.
